finality-aleph/src/aggregation/aggregator.rs

```diff
@@ -14,29 +14,112 @@ use std::{
     marker::PhantomData,
 };
 
-pub(crate) enum AggregatorError {
+pub enum AggregatorError {
+    LastHashPlaced,
+    NoHashFound,
+    DuplicateHash,
+}
+
+pub enum IOError {
     NetworkChannelClosed,
+    AggregatorError(AggregatorError),
```

Comment:
I don't think you need this, you actually handle all aggregator errors internally to `IO`, see a comment below for the single usage and why it should be removed.